src/lib/operators/projection.cpp

```diff
@@ -116,23 +127,27 @@ const std::string Projection::_get_type_of_expression(const std::shared_ptr<Expr
   const auto type_left = _get_type_of_expression(expression->left_child(), table);
   const auto type_right = _get_type_of_expression(expression->right_child(), table);
 
+  if (type_left == "NULL") return type_right;
+  if (type_right == "NULL") return type_left;
+
   // TODO(anybody): int + float = float etc...
   // This is currently not supported by `_evaluate_expression()` because it is only templated once.
-  Assert(type_left == type_right, "Projection currently only supports expressions with same type on both sides.");
+  Assert(type_left == type_right, "Projection currently only supports expressions with same type on both sides (" +
+                                      type_left + " vs " + type_right + ")");
   return type_left;
 }
 
 template <typename T>
-const tbb::concurrent_vector<T> Projection::_evaluate_expression(const std::shared_ptr<Expression>& expression,
-                                                                 const std::shared_ptr<const Table> table,
-                                                                 const ChunkID chunk_id) {
+const pmr_concurrent_vector<std::optional<T>> Projection::_evaluate_expression(
```

Comment:
could you maybe split this method? it is ~80 lines and covers quite a few different cases. at least the whole `is_literal` block should be extracted for readability.

Reply:
It is quite long, true. What do you mean by the "is_literal" block? Those are four different cases that have to be handled separately.

Having a hard time separating this into meaningful submethods. If they are only called in exactly one order and do not serve any purpose outside of this method, is that better than one single, long method?

Reply:
I guess this is more of a design preference question than anything else. So if you don't think it makes sense, then leave it as is. It's not a major deal to me.